static/js/chart/draw.ts

```diff
@@ -438,46 +469,85 @@ function drawLineChart(
 /**
  * Return all styles information for given dataGroupsDict.
  * {
- *     colors: [],
- *     dashes: [],
- *     statVars: [],
- *     geoIds: [],
- *     maxV: ,
+ *     colors: string[], an array of colors for statVars
+ *     dashes: string[], an array of dash styles for geoIds(if applied)
+ *     statVars: string[], an array of statVars
+ *     geoIds: string[], an array of geoIds
  * }
  *
  * @param dataGroupsDict
  */
-function generateStyles(
+function computePlotParams(
     dataGroupsDict: {[geoId: string]: DataGroup[] },
-) {
-  let styles = {};
-  styles["colors"] = [];
-  styles["dashes"] = [];
-  styles["statVars"] = [];
-  styles["geoIds"] = [];
+) : {
+  "colors": string[],
```

Comment:
This looks like a good case for an interface:
https://www.typescriptlang.org/docs/handbook/interfaces.html

It should simplify this chunk here a lot, where you have to redefine the type multiple times.

Reply:
Thanks for this suggestion, done~